paasta_tools/kubernetes_tools.py

```diff
@@ -1155,7 +1160,7 @@ def get_pod_template_spec(
                     service_namespace_config=service_namespace_config,
                 ),
                 share_process_namespace=True,
-                node_selector=self.get_node_selector(),
+                affinity=V1Affinity(node_affinity=self.get_node_affinity(),),
```

Comment:
Could we use node_selector for pool lable and use affinity for others, such as whitelist, blacklist and constraint?

Reply:
I'll move the pool affinity back to a selector. As a side note, constraints as affinities might not be a good fit, but we can discuss more in another PR.